subprojects/core/src/main/java/org/gradle/api/tasks/bundling/AbstractArchiveTask.java

```diff
@@ -467,6 +469,28 @@ public void setPreserveFileTimestamps(boolean preserveFileTimestamps) {
         archivePreserveFileTimestamps.set(preserveFileTimestamps);
     }
 
+    /**
+     * Specifies lastModified timestamp of entries in archive in milliseconds to epoch time.
+     * <p>
+     * If <tt>-1L</tt> there is no output timestamp specified.
+     * </p>
+     *
+     * @return milliseconds to epoch time if file timestamps should be set for archive entries
+     */
+    @Input
+    public long getOutputTimestamp() {
+        return outputTimestamp.get();
+    }
```

Comment:
Should the property itself be exposed rather than `getter/setter`?
It would avoid awkward cases like `-1`